pkg/core/blockchain.go

```diff
@@ -906,6 +919,72 @@ func (bc *Blockchain) storeBlock(block *block.Block) error {
 	return nil
 }
 
+func (bc *Blockchain) processTransfer(cache *dao.Cached, from util.Uint160, tx *transaction.Transaction, b *block.Block,
+	totals map[util.Uint256]util.Fixed8) error {
+
+	fromIndex, err := cache.GetNextTransferBatch(from)
+	if err != nil {
+		return err
+	}
+	for i := range tx.Outputs {
+		if !tx.Outputs[i].AssetID.Equals(GoverningTokenID()) && !tx.Outputs[i].AssetID.Equals(UtilityTokenID()) {
+			continue
+		}
+		if !from.Equals(tx.Outputs[i].ScriptHash) {
+			tr := &state.Transfer{
+				AssetID:   tx.Outputs[i].AssetID,
+				From:      from,
+				To:        tx.Outputs[i].ScriptHash,
+				Amount:    int64(tx.Outputs[i].Amount),
+				Block:     b.Index,
+				Timestamp: b.Timestamp,
+				Tx:        tx.Hash(),
+			}
+			isBig, err := cache.AppendTransfer(from, fromIndex, tr)
+			if err != nil {
+				return err
+			}
+			if isBig {
+				fromIndex++
+			}
+
+			index, err := cache.GetNextTransferBatch(tx.Outputs[i].ScriptHash)
+			if err != nil {
+				return err
+			}
+			isBig, err = cache.AppendTransfer(tx.Outputs[i].ScriptHash, index, tr)
+			if err != nil {
+				return err
+			}
+			if isBig {
+				if err := cache.PutNextTransferBatch(tx.Outputs[i].ScriptHash, index+1); err != nil {
+					return err
+				}
+			}
+		}
+		totals[tx.Outputs[i].AssetID] -= tx.Outputs[i].Amount
+	}
+	for asset, amount := range totals {
+		if amount.CompareTo(0) > 0 {
+			isBig, err := cache.AppendTransfer(from, fromIndex, &state.Transfer{
+				AssetID:   asset,
+				From:      from,
```

Comment:
You're adding this to `from`, but C# also adds `received` event for (virtual) account zero. It allows for token burns tracking.

Reply:
Fixed.